core/jobs_test.py

```diff
@@ -363,54 +363,64 @@ def test_cancelling_one_unfinished_job(self):
         self.assertIsNone(DummyJobManager.get_error(job2_id))
 
     def test_compress_output_list_with_single_char_outputs(self):
-        sample_output_list = [1, 2, 3, 4, 5]
-        expected = ['1', '2', '3', '<TRUNCATED>']
-        actual = jobs.BaseJobManager._compress_output_list(  # pylint: disable=protected-access
-            sample_output_list, test_only_max_output_len=3)
-        self.assertEquals(expected, sorted(actual))
+        input_list = [1, 2, 3, 4, 5]
+        expected_output_possibilities = [
+            {'1', '2', '3', '<TRUNCATED>'}, {'1', '2', '4', '<TRUNCATED>'},
+            {'1', '2', '5', '<TRUNCATED>'}, {'1', '3', '4', '<TRUNCATED>'},
+            {'1', '3', '5', '<TRUNCATED>'}, {'1', '4', '5', '<TRUNCATED>'},
+            {'2', '3', '4', '<TRUNCATED>'}, {'2', '3', '5', '<TRUNCATED>'},
+            {'2', '4', '5', '<TRUNCATED>'}, {'3', '4', '5', '<TRUNCATED>'},
+        ]
+        actual_output = jobs.BaseJobManager._compress_output_list(  # pylint: disable=protected-access
+            input_list, test_only_max_output_len=3)
+        self.assertIn(set(actual_output), expected_output_possibilities)
 
     def test_compress_output_list_with_multi_char_outputs(self):
-        sample_output_list = ['abcd', 'efgh', 'ijkl']
-        expected = ['abcd', 'efgh', 'ij <TRUNCATED>']
-        actual = jobs.BaseJobManager._compress_output_list(  # pylint: disable=protected-access
-            sample_output_list, test_only_max_output_len=10)
-        self.assertEquals(expected, sorted(actual))
+        input_list = ['abcd', 'efgh', 'ijkl']
+        expected_output_possibilities = [
+            {'abcd', 'efgh', 'ij <TRUNCATED>'},
+            {'abcd', 'ijkl', 'ef <TRUNCATED>'},
+            {'efgh', 'ijkl', 'ab <TRUNCATED>'},
+        ]
+        actual_output = jobs.BaseJobManager._compress_output_list(  # pylint: disable=protected-access
+            input_list, test_only_max_output_len=10)
+        self.assertIn(set(actual_output), expected_output_possibilities)
 
     def test_compress_output_list_with_zero_max_output_len(self):
-        sample_output_list = [1, 2, 3]
-        expected = ['<TRUNCATED>']
-        actual = jobs.BaseJobManager._compress_output_list(  # pylint: disable=protected-access
-            sample_output_list, test_only_max_output_len=0)
-        self.assertEquals(expected, sorted(actual))
+        input_list = [1, 2, 3]
+        expected_output = {'<TRUNCATED>'}
+        actual_output = jobs.BaseJobManager._compress_output_list(  # pylint: disable=protected-access
+            input_list, test_only_max_output_len=0)
+        self.assertEquals(expected_output, set(actual_output))
```

Comment:
I'm not sure I see the point of the set here. I think a list and direct comparison works fine; ditto below.

(On the other hand, I do agree with the naming changes and the consistency that that introduces -- that is nice.)